routes/user.js

```diff
@@ -76,28 +82,260 @@ router.post('/library', function(req, res, next) {
         })
 })
 
-router.get('/playlist', function(req, res, next){
-  Playlist.find({owner: req.user._id})
-  .populate('songs')
-  .exec(function(err, playlists){
-      if (err) return handleError(err)
-      // console.log(playlists)
-      res.render('playlist', {playlists: playlists})
-  })
+router.get('/playlist', function(req, res, next) {
+    Playlist.find({
+            owner: req.user._id
+        })
+        .populate('songs')
+        .exec(function(err, playlists) {
+            if (err) return handleError(err)
+            res.render('playlist', {
+                playlists: playlists
+            })
+        })
 })
 
-router.post('/playlist', function(req, res, next){
-  var name = req.body.name
-  Playlist.findOne({owner: req.user._id, name: name})
-  .populate('songs')
-  .exec(function(err, playlist){
-    // console.log(JSON.stringify(playlist)
-    if (err) return handleError(err)
-    // var titles = playlist.songs.map((s) => s.title)
-    // var songs = playlist.map((s) => s.songs)
-    // console.log(songsTitle)
-    res.send({songs: playlist.songs})
-  })
+router.post('/playlist', function(req, res, next) {
+    var name = req.body.name
+    Playlist.findOne({
+            owner: req.user._id,
+            name: name
+        })
+        .populate('songs')
+        .exec(function(err, playlist) {
+            // console.log(JSON.stringify(playlist)
+            if (err) return handleError(err)
+                // var titles = playlist.songs.map((s) => s.title)
+                // var songs = playlist.map((s) => s.songs)
+                // console.log(songsTitle)
+            res.send({
+                songs: playlist.songs,
+                name: playlist.name
+            })
+        })
+})
+
+router.get('/playlist/:playlist_name/export1', function(req, res, next) {
+    var playlistName = req.params.playlist_name
+    var translationss = []
+    Playlist.findOne({
+            owner: req.user._id,
+            name: playlistName
+        })
+        .populate('songs')
+        .exec(function(err, playlist) {
```

Comment:
a bit confused about this code block. 
1. Is it creating a song if it doesn't exist?
2. It is rendering a page.

What is it trying to do and should it be broken up into separate calls or modules?